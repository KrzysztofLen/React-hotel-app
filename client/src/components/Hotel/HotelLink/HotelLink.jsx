// @flow
import React from 'react';
import {NavLink} from 'react-router-dom';

type Props = {
	id: number,
	hotelName: string
}

const HotelLink = (data: Props) => (
	<div className="hotel__details--name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
	</div>
);

export default HotelLink;