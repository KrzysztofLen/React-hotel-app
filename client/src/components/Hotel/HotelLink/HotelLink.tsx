import React from 'react';
import {NavLink} from 'react-router-dom';

interface IProps {
	id: number,
	hotelName: string
}

const HotelLink = (data: IProps) => (
	<div className="hotel__name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
	</div>
);

export default HotelLink;