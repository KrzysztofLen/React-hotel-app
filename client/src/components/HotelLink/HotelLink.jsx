import React from 'react';
import {NavLink} from 'react-router-dom';

const HotelLink = (data) => (
	<div className="hotel__details--name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
		<div className="hotel__details--adressBox">
			<span className="hotel__details--adress">{data.hotelAdress}</span>
			<p className="hotel__details--distance">(<span className="hotel__details--distanceValue">{data.hotelDistance}</span> km from center)</p>
		</div>
	</div>
);

export default HotelLink;