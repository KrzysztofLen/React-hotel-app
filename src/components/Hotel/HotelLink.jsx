import React from 'react';
import {NavLink} from 'react-router-dom';

const HotelLink = (data) => (
	<div className="hotel__details--name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
	</div>
);

export default HotelLink;