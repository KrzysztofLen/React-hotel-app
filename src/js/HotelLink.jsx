import React, {Component} from 'react';
import {NavLink} from 'react-router-dom';

const HotelLink = (data) => <NavLink to={"/question/" + data.id} className="question">{data.hotelName}</NavLink>;

export default HotelLink;