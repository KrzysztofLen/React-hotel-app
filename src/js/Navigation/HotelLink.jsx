import React from 'react';

export const HotelLink = () => (
	<li className="side-nav__item side-nav__item--active">
		<a href="#" className="side-nav__link">
			<span>Hotel's</span>
		</a>
	</li>
);