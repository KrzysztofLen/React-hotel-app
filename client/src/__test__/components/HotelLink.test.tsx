import React from 'react';
import { shallow } from 'enzyme';
import HotelLink from '../../components/Hotel/HotelLink/HotelLink';
import { shallowToJson } from 'enzyme-to-json';

const app_data = [
  {
    id: 0,
    name: 'Test Eva',
    hotel_name: 'Test Sunset Willow Resort',
    image: 'https://preview.ibb.co/ivpH07/alex_block_354270_unsplash.jpg',
    rate: 5,
    more_activities: true,
    activities_number: 6,
    commented: 5,
    peers: 46,
    discussions: 29,
    findings: 19,
    questions: 10,
    activity_level: 2,
  },
];

describe('[COMPONENT] HotelLink', () => {
  it('should render HotelLink correctly', () => {
    const wrapper = shallow(
      <HotelLink hotelName={app_data[0].hotel_name} id={app_data[0].id} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
