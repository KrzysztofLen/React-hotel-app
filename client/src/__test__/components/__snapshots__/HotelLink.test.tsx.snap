// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] HotelLink should render HotelLink correctly 1`] = `
<div
  className="hotel__name"
>
  <NavLink
    className="hotel__link"
    to="/hotel/0"
  >
    Test Sunset Willow Resort
  </NavLink>
</div>
`;
